### ISSUE TITLE
'node examples/http.js 8080' crashing on connect

### ISSUE THREAD

**Issue by 123batman** [amark/gun#332]
Hello!
There seems to be a bug in the default gun server. Or am I doing something wrong?
```
username@usernamePC:~/sync/code/planner/node_modules/gun$ npm start

> gun@0.6.3 start /home/username/sync/code/planner/node_modules/gun
> node examples/http.js 8080

Hello wonderful person! :) Thanks for using GUN, feel free to ask for help on https://gitter.im/amark/gun and ask StackOverflow questions tagged with 'gun'!
WARNING! This `file.js` module for gun is intended for local development testing only!
Server started on port 8080 with /gun
/home/username/sync/code/planner/node_modules/gun/node_modules/ws/lib/Receiver.js:294
  if (mask != null && buf != null) bufferUtil.unmask(buf, mask);
                                             ^

TypeError: Cannot read property 'unmask' of undefined
    at Receiver.unmask (/home/username/sync/code/planner/node_modules/gun/node_modules/ws/lib/Receiver.js:294:46)
    at Receiver.finish (/home/username/sync/code/planner/node_modules/gun/node_modules/ws/lib/Receiver.js:582:19)
    at Receiver.expectHandler (/home/username/sync/code/planner/node_modules/gun/node_modules/ws/lib/Receiver.js:570:33)
    at Receiver.add (/home/username/sync/code/planner/node_modules/gun/node_modules/ws/lib/Receiver.js:95:24)
    at Socket.realHandler (/home/username/sync/code/planner/node_modules/gun/node_modules/ws/lib/WebSocket.js:800:20)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:189:7)
    at readableAddChunk (_stream_readable.js:176:18)
    at Socket.Readable.push (_stream_readable.js:134:10)
    at TCP.onread (net.js:551:20)

npm ERR! Linux 4.4.0-65-generic
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "start"
npm ERR! node v7.6.0
npm ERR! npm  v4.3.0
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! gun@0.6.3 start: `node examples/http.js 8080`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the gun@0.6.3 start script 'node examples/http.js 8080'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the gun package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node examples/http.js 8080
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs gun
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls gun
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
```

**Comment by amark** [comment]
Oh dang! That is annoying. I apologize. I recently updated the `ws` version, and it looks like something is crashing. I have a prototype for `uws` which is much better/faster.

Do you get the same error if you re-run it?

It looks like it only crashes when you receive data, is it possible the data is malformed when sent? (I don't know how it would be possible to malform data in JS though, lol).

**Comment by swhgoon** [comment]
I'm using `"ws": "^1.1.2"` which runs OK

**Comment by 123batman** [comment]
I tried to isolate the bug but it is somewhat difficult. The problem does not seem to lie in the Gun client. I commented out almost all of the gun related stuff and stepped through the remaining calls involving gun with the debugger. All that is left:
```
import Gun from 'gun';
console.log(Gun)
const gun = Gun('http://localhost:8080/gun');//
gun.get('taskRandomID').put({ taskID: 'randomTaskID', title: 'test'})
```
I am using Gun inside the meteor framework and can only assume something weird is going on with Websockets in the background. Either way I don't think anything should be able to crash the Gun server like that. Maybe the bug will be gone with your uws code. When do you want to publish it? :)

Until then I am going to continue developing with local gun only.

Thanks, swhgoon. I tried that as well. Unfortunately it does not work for me.

**Comment by amark** [comment]
if you are using import on the client, you probably need to do `import Gun from 'gun/gun'` instead.

Just published `0.6.4` which has the high performance uWS load testing in it. However to start a server you now need to do (server side) `Gun({web: server})` not `gun.wsp(server)` as before. 

If you are using meteor also, are they both running on the same websocket server? I could see how that could mess things up maybe. I'm not sure how Meteor works, but possibly?

**Comment by amark** [comment]
@123batman haven't heard from you, and nobody else (which isn't a good metric, admittedly) has had an issue with the example server crashing on connect and we're now at `0.9.4`. So I'm gonna close, please re-open if you can figure out how to reproduce. Sorry, cheers!
